OneCommunityGlobal/HighestGoodNetworkApp
# Anand_BadgeManagement_Unit_Test

# Description
Tested badge validation and assignment logic - including Redux actions, API calls, error handling, and success messaging.

Test case file: `src\components\Badge\__tests__\BadgeManagement.test.js`
Reference: `src\actions\badgeManagement.js`

## Related PRS (if any):
none

## Main changes explained:
`validateBadges` function:
- Tested validation logic for missing first and last names, ensuring appropriate error messages are shown when necessary.
- Verified that no actions are triggered if both names are provided.

`assignBadges` function:
- Checked error handling when no badges are selected, ensuring the appropriate error message is displayed.
- Verified API call to fetch the user profile by name and handled the case where the user is not found.
- Tested successful badge assignment by confirming that a PUT request is made to the server and appropriate success messages are displayed.
- Ensured proper error handling and messaging when server issues occur during the badge assignment process.

## How to test:
1.Check into current branch
2. Do npm install
3. Do npm test -- BadgeManagement.test.js to run the test
4. Verify that all test cases pass successfully

## Screenshots or videos of changes:

![BadgeManagement unit tests](https://github.com/user-attachments/assets/9643bf6f-e56e-4f07-b419-fee9e745728f)
validateBadges:
![validateBadge](https://github.com/user-attachments/assets/29a5e6fc-73a1-4c40-8625-2c3762a8f9e4)
assignBadges:
![assignBadge](https://github.com/user-attachments/assets/19026b6d-d4a6-4e87-a135-2a94020f93de)
